lib/srv/desktop/tdp/proto.go

```diff
@@ -440,19 +444,21 @@ func (r ClientUsername) Encode() ([]byte, error) {
 func decodeClientUsername(in io.Reader) (ClientUsername, error) {
 	username, err := decodeString(in, windowsMaxUsernameLength)
 	if err != nil {
+		if errors.Is(err, trace.LimitExceeded(stringMaxLenErrMsg)) {
+			// Change the error message here so it's considered a fatal error
+			return ClientUsername{}, trace.LimitExceeded("ClientUsername exceeded maximum length")
+		}
 		return ClientUsername{}, trace.Wrap(err)
 	}
 	return ClientUsername{Username: username}, nil
 }
 
+// Error is a tdp Error. It has been deprecated
```

Comment:
Is this going to be backported to v11 or only v12 and up?

If we know it's only v12 and up, maybe we mention that here?

```
// Error is used to send a fatal error message to the browser.
// In Teleport 12 and up, Error is deprecated and Notification
// should be preferred.
```

Reply:
This is only 12 and up

> On Teleport major version releases, we instruct customers to update proxy/auth before w_d_s, meaning that we can add TDP messages that are sent from server (w_d_s) to client (proxy) without issue (as is the case here). For this reason, these changes won't be backported, and will just be included in the Teleport 12 release.